docs/100-upgrade/migration-guides/3.2-3.3.mdx

```diff
@@ -0,0 +1,54 @@
+---
+title: 3.2 -> 3.3
+sidebar_position: 98 # decrease for each release
+description:
+  This page lists the highlights for upgrading a project from Front-Commerce 3.2
+  to 3.3.
+---
+
+<p>{frontMatter.description}</p>
+
+## Update dependencies
+
+Update all your `@front-commerce/*` dependencies to this version:
+
+```shell
+pnpm update "@front-commerce/\*@3.3.0"
+```
+
+## Code changes
+
+### Automatic session cookie headers
+
+In this release, we added a feature that will automatically set the `Set-Cookie`
+header for the session cookie when the user session needs to be commited.
+
+:::caution
+
+Due to how the requests are handled by the current version of Remix, this will
+only work when creating response with `json()` without passing a second
+parameter.
+
+:::
+
+```diff title="app/routes/my-route.ts"
+import { FrontCommerceApp } from "@front-commerce/remix";
+import { json } from "@front-commerce/remix/node";
+import { MyQuery } from "~/graphql/graphql";
+
+const loader = async ({ context, request }) => {
+    const app = new FrontCommerceApp(context.frontCommerce);
+
+    const data = app.query(MyQuery);
+-    return json(data, {
+-        headers: {
+-            "Set-Cookie": await app.user.session.commit(),
+-        },
+-    });
++    return json(data);
+}
+```
+
+This feature includes a change on `entry.server.ts`. If you changed this file in
+your project, you will need to adapt it to take the new feature into account
+([related commit](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/commit/55a618228874d9213b278faf8e742fd9095f40d4)).
```

Comment:
**issue:** this must be outside of this section. Entrypoints and server are global (i.e: transformers could do many things).

**suggestion:** similar to [3.1 -> 3.2](https://developers.front-commerce.com/docs/3.x/upgrade/migration-guides/3.1-3.2#remix-entrypoints), start the migration guide with the required update  to this file.\
Embed the diff directly here, to prevent an extra redirection.

Reply:
Added diffs in `details` at the beginning of the guide (effectively moving down the `caution` to the bottom).